corehq/apps/app_manager/static/app_manager/js/download_async_modal.js

```diff
@@ -13,65 +13,81 @@ $(function(){
         self.$downloading = self.$el.find("#" + self.el_id + "-downloading");
 
+        self.download_in_progress = false;
+        self.download_poll_url = null;
+        self.download_poll_id = null;
+
         self.init = function(){
+            self.download_in_progress = false;
+            self.download_poll_url = null;
+            self.download_poll_id = null;
             self.$download_progress.addClass("hide");
             self.$downloading.removeClass("hide");
         };
 
-        self.startPollDownloadStatus = function(data){
-            var keep_polling = true;
-            var pollDownloadStatus = function(){
-                if (keep_polling) {
-                    $.ajax({
-                        url: data.download_url,
-                        success: function (resp) {
-                            if (resp.trim().length) {
-                                self.$downloading.addClass("hide");
-                                self.$download_progress.html(resp).removeClass("hide");
-                                if (self.$download_progress.find(".alert-success").length) {
-                                    keep_polling = false;
-                                }
-                            }
-                            if (keep_polling) {
-                                setTimeout(pollDownloadStatus, self.POLL_FREQUENCY);
-                            }
-                        },
-                        error: function (resp) {
-                            self.downloadError(resp.responseText);
+        self.pollDownloadStatus = function(){
+            if (self.download_in_progress) {
+                $.ajax({
+                    url: self.download_poll_url,
+                    success: function (resp) {
+                        self.updateProgress(resp);
+                        if (!self.isDone(resp)) {
+                            setTimeout(self.pollDownloadStatus, self.POLL_FREQUENCY);
                         }
-                    });
-                }
-            };
-
+                    },
+                    error: function (resp) {
+                        self.downloadError(resp.responseText);
+                    }
+                });
+            }
+        };
 
-            self.$el.on("hidden hidden.bs.modal", function(){
-                keep_polling = false;
-                self.init();
-            });
+        self.updateProgress = function (progress_response) {
+            if (progress_response.trim().length) {
+                self.$downloading.addClass("hide");
+                self.$download_progress.html(progress_response).removeClass("hide");
+            }
+        };
 
-            pollDownloadStatus();
+        self.isDone = function (progress_response) {
+            var ready_id = 'ready_' + self.download_poll_id,
+                error_id = 'error_' + self.download_poll_id;
+            return !!progress_response &&
```

Comment:
I don't see `_.any` in the Underscore.js docs.

True that I don't really need the !! since I'm just checking for *truthy*. I guess I could rewrite the test to split the true and false expectations.